tools/harness-thci/OpenThread_BR.py

```diff
@@ -618,31 +618,74 @@ def __dumpSyslog(self):
     def mdns_query(self, dst='ff02::fb', service='_meshcop._udp.local', addrs_blacklist=[]):
         print('mdns_query %s %s %s' % (dst, service, addrs_blacklist))
 
-        result = self.bash('dig -p 5353 @%s %s ptr +time=2 +retry=2' % (dst, service))
-
+        # For BBR-TC-03 or DH test cases just send a query
         if dst == 'ff02::fb' and not addrs_blacklist:
-            return (0, '')
-
-        # Remove responses from unwanted devices
-        responses = ' '.join(result).split(';; ANSWER SECTION:')
-        for response in responses:
-            if not set(response.split()).isdisjoint(set(addrs_blacklist)):
-                break
-
-        # Records patterns:
-        # raspberrypi-2.local.	10	IN	AAAA	fe80::81:46ff:fe0d:bfe.43684
-        # OpenThread_BorderRouter._meshcop._udp.local. 10\tIN SRV 0 0 49153 raspberrypi.local.
-        try:
-            addr = response.split('AAAA\tfe80')[1].split(' ')[0]
-            addr = 'fe80%s%%eth0' % addr
-        except Exception:
-            raise Exception('Unable to find the DUT address in the mDNS response')
-        try:
-            port = int(response.split('IN SRV')[1].split(' ')[3])
-        except Exception:
-            raise Exception('Unable to find the DUT port in the mDNS response')
+            self.bash('dig -p 5353 @%s %s ptr' % (dst, service))
+            return
 
-        return (addr, port)
+        # For MATN-TC-17 and MATN-TC-18 use Zeroconf to get the BBR address and border agent port
+        from zeroconf import ServiceBrowser, ServiceStateChange, Zeroconf, DNSAddress, DNSService, DNSText
+
+        def on_service_state_change(zeroconf, service_type, name, state_change):
+            if state_change is ServiceStateChange.Added:
+                zeroconf.get_service_info(service_type, name)
+
+        class BorderAgent:
```

Comment:
```suggestion
        class BorderAgent(object):
```
nit: should inherit from object in PY2